packages/server/graphql/intranetSchema/mutations/runScheduledJobs.ts

```diff
@@ -35,49 +30,62 @@ const processMeetingStageTimeLimits = async (job: ScheduledJobMeetingStageTimeLi
       .nth(0)
       .default(null) as unknown) as SlackAuth
   }).run()
+  return {slackNotification, slackAuth}
+}
 
-  let sendViaSlack = Boolean(slackAuth?.botAccessToken && slackNotification?.channelId)
-  if (sendViaSlack) {
-    const {channelId} = slackNotification
-    if (!channelId) {
-      sendViaSlack = false
-    } else {
-      const {botAccessToken} = slackAuth
-      const manager = new SlackServerManager(botAccessToken)
-      const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`)
-      const slackText = `Time’s up! Advance your meeting to the next phase: ${meetingUrl}`
-      const res = await manager.postMessage(channelId, slackText)
-      if (!res.ok) {
-        sendViaSlack = false
-      }
-    }
-  }
-  if (!sendViaSlack) {
-    const notification = new NotificationMeetingStageTimeLimitEnd({
-      meetingId,
-      userId: facilitatorUserId
-    })
-    await r
-      .table('Notification')
-      .insert(notification)
-      .run()
-    publish(SubscriptionChannel.NOTIFICATION, facilitatorUserId, 'MeetingStageTimeLimitPayload', {
-      notification
-    })
-  }
-
+const processMeetingStageTimeLimits = async (
+  job: ScheduledJobMeetingStageTimeLimit,
+  {dataLoader}: {dataLoader: DataLoaderWorker}
+) => {
   // get the meeting
   // get the facilitator
   // see if the facilitator has turned on slack notifications for the meeting
-  // if so, send the facilitator a slack notification
-  // if not, send the facilitator an in-app notification
+  // detect integrated services
+  // if slack, send slack
+  // if mattermost, send mattermost
+  // if no integrated notification services, send an in-app notification
+  const {meetingId} = job
+  const meeting = await dataLoader.get('newMeetings').load(meetingId)
+  const {teamId, facilitatorUserId} = meeting
+  const [{slackNotification, slackAuth}, {mattermostAuth}] = await Promise.all([
+    getSlackNotificationAndAuth(teamId, facilitatorUserId),
+    dataLoader.get('mattermostBestAuthByUserIdTeamId').load({userId: facilitatorUserId, teamId})
+  ])
+  const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`)
+
+  const sendViaMattermost = mattermostAuth?.isActive
```

Comment:
`isActive` is guaranteed to be true, so the following will always be true (i think the object destructure up above ruined the strong types. noImplicitAny will save us all soon 😄 )

```
  const sendViaMattermost = !!mattermostAuth
```